src/types.rs

```diff
@@ -1,5 +1,65 @@
 use std::collections::HashMap;
 
+use std::fmt;
+use std::cmp;
+use serde::*;
```

Comment:
Is there a good reason we need a glob import here? I generally try to avoid these as you may pull in something you don't actually need and it confusingly conflicts with something else you try to do later on. If there's not too many types, it's better to be explicit.

Reply:
Thanks for fixing all my nits! If you're willing to do it, just go ahead and tidy up everything like this. There's no good reason for the formatting and glob import, they're remnants from when I copy+pasted stuff. I wanted to run rustfmt, but it ended up reformatting about half the lines in the file.